### PR TITLE
Issue 47735: QWP date filter not being formatted

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Data Region loading async, like any QWP, has format issues on date column filters. See related ticket for repro. This PR formats the date filter values for async load the same way it is formatted for the non-async load.

https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=47735

#### Changes
* Add date format to DataRegion async load
* Add QWP test

### CODE REVIEW

**Review by labkey-jeckels — APPROVED** [review]
Since this is copying similar code, I assume we need to have this hard-coded date format instead of respecting the server's configured format?

Can we pull this into a small `formatDate()` function instead of duplicating it?

File: api/webapp/clientapi/dom/DataRegion.js [hunk 1]
```diff
@@ -3814,7 +3814,19 @@ if (!LABKEY.DataRegions) {
                     else if (!LABKEY.Utils.isArray(params[pair[0]])) {
                         params[pair[0]] = [params[pair[0]]];
                     }
-                    params[pair[0]].push(pair[1]);
+
+                    var value = pair[1];
+
+                    // This needs to be formatted for the response passed back to the grid for the filter display and
+                    // filter dialog to render correctly
+                    if (LABKEY.Utils.isDate(value)) {
```
Comment (labkey-nicka) [inline api/webapp/clientapi/dom/DataRegion.js]: Definitely agree with @labkey-jeckels comment to consolidate this code to share with logic currently inside `_buildQueryString`.

File: api/webapp/clientapi/dom/DataRegion.js [hunk 2]
```diff
@@ -3814,7 +3814,19 @@ if (!LABKEY.DataRegions) {
                     else if (!LABKEY.Utils.isArray(params[pair[0]])) {
                         params[pair[0]] = [params[pair[0]]];
                     }
-                    params[pair[0]].push(pair[1]);
+
+                    var value = pair[1];
+
+                    // This needs to be formatted for the response passed back to the grid for the filter display and
```
Comment (labkey-nicka) [inline api/webapp/clientapi/dom/DataRegion.js:3827]: nit: Please add the issue here for context and dynamic linking. 

```js
// Issue 47735: This needs to be formatted for the response passed back...
```

File: api/webapp/clientapi/dom/DataRegion.js [hunk 3]
```diff
@@ -3814,7 +3814,19 @@ if (!LABKEY.DataRegions) {
...
-                    params[pair[0]].push(pair[1]);
+
+                    var value = pair[1];
+
+                    // This needs to be formatted for the response passed back to the grid for the filter display and
+                    // filter dialog to render correctly
+                    if (LABKEY.Utils.isDate(value)) {
+                        value = $.format.date(value, 'yyyy-MM-dd');
+                        if (LABKEY.Utils.endsWith(value, 'Z')) {
+                            value = value.substring(0, value.length - 1);
```
Comment (labkey-nicka) [inline api/webapp/clientapi/dom/DataRegion.js]: Consider adding a regression test case for this issue to `QWPDemo.js` and the associated `DataRegionTest`.

**Review by labkey-nicka — APPROVED** [review]

File: api/webapp/clientapi/dom/DataRegion.js (hunk 1 above)
Comment (labkey-martyp) [reply api/webapp/clientapi/dom/DataRegion.js]: refactor done

File: api/webapp/clientapi/dom/DataRegion.js (hunk 2 above)
Comment (labkey-martyp) [reply api/webapp/clientapi/dom/DataRegion.js:3827]: added

File: api/webapp/clientapi/dom/DataRegion.js (hunk 3 above)
Comment (labkey-martyp) [reply api/webapp/clientapi/dom/DataRegion.js]: done

**Pull request merged**